cloud/deploy/deploy.go

```diff
@@ -59,84 +65,148 @@ type deploymentInfo struct {
 	webserverURL   string
 }
 
-// Deploy pushes a new docker image
-func Deploy(path, deploymentID, wsID, pytest, envFile, imageName string, prompt bool, client astro.Client) error {
-	// Get cloud domain
-	c, err := config.GetCurrentContext()
+func deployDags(path, domain string, deployInfo *deploymentInfo, client astro.Client) error {
+	fmt.Println("Initiating DAGs Deployment for: " + deployInfo.deploymentID)
+	dagDeployment, err := deployment.Initiate(deployInfo.deploymentID, client)
 	if err != nil {
 		return err
 	}
 
-	cloudDomain := c.Domain
-	if cloudDomain == "" {
-		return errors.New("no domain set, re-authenticate")
-	}
+	// Check the dags directory
+	dagsPath := path + "/dags"
 
-	deployInfo, err := getDeploymentInfo(deploymentID, wsID, prompt, cloudDomain, client)
+	// Generate the dags tar
+	err = fileutil.Tar(dagsPath, "/tmp")
 	if err != nil {
 		return err
 	}
 
-	// Build our image
-	version, err := buildImage(&c, path, deployInfo.currentVersion, deployInfo.deployImage, imageName, client)
+	sasDagClient, err := azure.CreateSASDagClient(dagDeployment.DagURL)
 	if err != nil {
 		return err
 	}
 
-	err = parseDAG(pytest, version, envFile, deployInfo.deployImage, deployInfo.namespace)
+	dagFile, err := os.Open(dagsFile)
 	if err != nil {
 		return err
 	}
-
-	// Create the image
-	imageCreateInput := astro.ImageCreateInput{
-		Tag:          version,
-		DeploymentID: deployInfo.deploymentID,
-	}
-	imageCreateRes, err := client.CreateImage(imageCreateInput)
+	versionID, err := azure.Upload(sasDagClient, dagFile)
 	if err != nil {
 		return err
 	}
 
-	domain := c.Domain
-	if strings.Contains(domain, "cloud") {
-		splitDomain := strings.SplitN(domain, ".", splitNum) // This splits out 'cloud' from the domain string
-		domain = splitDomain[1]
+	// Delete the temporary tar file
+	err = os.Remove(dagsFile)
+	if err != nil {
+		return err
```

Comment:
```suggestion
	// Delete the temporary tar file
	defer func() {
	    err = os.Remove(dagsFile)
	    if err != nil {
	        fmt.Println("Failed to delete temporary tar file: %s", err.Error())
	    }
	}
```
We might want to delete the file as part of defer so that we still delete the files even if there is any panic or failure on `Upload`.
You can add this right after the tar file is created, which I think is here if I am not wrong https://github.com/astronomer/astro-cli/pull/624/files#diff-05e6dbfa75ff6117dd1c27f366e81a20392fd867e25cbeb5693aa1224bdcf7ffR83

Reply:
yeah, this makes sense. But we need to that post upload operation.